gettsim/policy_for_date.py

```diff
@@ -19,55 +19,101 @@
+            tax_data["calc_hhfreib"] = calc_hhfreib_until2014
+        else:
+            tax_data["calc_hhfreib"] = calc_hhfreib_from2015
+        if year >= 2010:
+            tax_data["vorsorge"] = vorsorge2010
+        else:
+            tax_data["vorsorge"] = vorsorge_dummy
+
+        # TODO: We need to adapt favorability check for that. See
+        #  https://github.com/iza-institute-of-labor-economics/gettsim/issues/81 for
+        #  details.
+        # if year >= 2009:
+        #     tax_data["zve_list"] = ["nokfb", "kfb", "abg_nokfb", "abg_kfb"]
+        # else:
+        #     tax_data["zve_list"] = ["nokfb", "kfb"]
+        tax_data["zve_list"] = ["nokfb", "kfb"]
+
+    elif group == "kindergeld":
+        if year > 2011:
+            tax_data["childben_elig_rule"] = kg_eligibility_hours
+        else:
+            tax_data["childben_elig_rule"] = kg_eligibility_wage
+
+    elif group == "wohngeld":
+        if year < 2009:
+            tax_data["calc_max_rent"] = calc_max_rent_until_2008
+        else:
+            tax_data["calc_max_rent"] = calc_max_rent_since_2009
+
+    elif group == "e_st":
+        tax_data["tax_schedule"] = tarif
+
+    elif group == "ges_renten_vers":
+        if year > 2017:
+            tax_data["calc_rentenwert"] = _rentenwert_from_2018
+        else:
+            tax_data["calc_rentenwert"] = _rentenwert_until_2017
+
+    return tax_data
+
+
+def load_ordanary_data_group(tax_data_raw, actual_date):
     tax_data = {}
-    this_year = datetime.date(year=year, month=month, day=day)
     for key in tax_data_raw:
-        if tax_data_raw[key]["values"] is not None:
-            policy_dates = tax_data_raw[key]["values"]
-            past_policies = [x for x in policy_dates if x <= this_year]
-            if not past_policies:
+        policy_dates = tax_data_raw[key]["values"]
+        past_policies = [x for x in policy_dates if x <= actual_date]
+        if not past_policies:
+            # TODO: Should there be missing values or should the key not exist?
+            tax_data[key] = np.nan
+        else:
+            policy_in_place = np.max(past_policies)
+            tax_data[key] = tax_data_raw[key]["values"][policy_in_place]["value"]
+    return tax_data
+
+
+def load_ges_renten_vers_params(raw_pension_data, actual_date):
+    pension_data = {}
+    # meanwages is only filled until 2016. The same is done in the pension function.
+    min_year = min(actual_date.year, 2016)
+    for key in raw_pension_data:
+        data_years = list(raw_pension_data[key]["values"])
+        # For calculating pensions we need demographic data up to three years in the
+        # past.
+        for year in range(min_year - 3, min_year + 1):
+            past_data = [x for x in data_years if x.year <= year]
+            if not past_data:
                 # TODO: Should there be missing values or should the key not exist?
-                tax_data[key] = np.nan
+                pension_data[f"{key}_{year}"] = np.nan
```

Comment:
Why store this stuff with such weird keys and not in DataFrames?

Reply:
I was trying to avoid DataFrames because then I would need to use pandas?! I am really not sure about this and had my doubts as well. I thought this would be an elegant way...

Reply:
Sort-of third Normal Form for data management: "No structure in variable names" :-)

Reply:
Opened #116